safe-global/safe-wallet-web
# feat: Improve safe creation status screen

## What it solves

Resolves #3612

## How this PR fixes it

- Removes the timeout for safe creation transactions
- Only navigates to the status screen after the user submits the safe creation in their wallet
- Renames the Cancel button to `Back to homepage`
- Renames the Retry button to `Try again`
- Navigates the user to the safe creation review screen when they press `Try again`
- Adds a new screen for when the transaction has failed
- Removes the `StatusStepper`
- Removes `useSafeCreation`
- Removes `useSafeCreationEffects`
- Shows a speed-up message if the transaction takes longer than 15 seconds
- Removes `CreationModal` and instead shows the new Success screen with the address attached

## ToDos

- [x] Need to figure out how to select the correct pending safe creation as there can be two now with counterfactual
- [x] Still pass `safeViewRedirectURL` on SUCCESS
- [x] Navigate user to dashboard on SUCCESS
- [x] Check that analytics events are still mapped correctly
- [x] Add safe (+ owners) to address book on SUCCESS

## How to test it

- create a safe through a safe
- the user should only proceed to the status screen once they submit the tx in their wallet

## Screenshots
<img width="1203" alt="Screenshot 2024-05-30 at 16 24 29" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/8a6f4936-abc5-41e1-ab6d-711fe5fbeba0">
<img width="1215" alt="Screenshot 2024-05-30 at 16 31 20" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/0b1b9756-d5e4-46fb-b85c-960261b50537">
<img width="1209" alt="Screenshot 2024-05-30 at 16 36 44" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/6eb665db-8c2b-4475-978e-bc79581602ad">
<img width="639" alt="Screenshot 2024-06-18 at 10 40 50" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/bba583e5-97f7-4c9b-a1c6-d2ba3c1a4bd8">

https://github.com/safe-global/safe-wallet-web/assets/5880855/19a1c895-2749-4a7e-b077-86abb1ea15af


## Checklist
